### PR TITLE
Refactor VersionUtils and make the code thread-safe

### PR DESCRIPTION
**Pull request by stanculescu**

### CODE REVIEW

File: server.api/src/main/java/org/kaazing/gateway/server/impl/VersionUtils.java [hunk 1]
```diff
@@ -238,4 +203,35 @@ private static boolean getProductInfoFromJar(String pathEntry, Map<String, Attri
         }
         return result;
     }
+
+    private static class GatewayProductVersionType {
```
Comment (DoruM) [inline server.api/src/main/java/org/kaazing/gateway/server/impl/VersionUtils.java]: Rename this as ProductInfo. The class should have the members : title, version, edition, dependencies.

File: server.api/src/main/java/org/kaazing/gateway/server/impl/VersionUtils.java [hunk 2]
```diff
@@ -238,4 +203,35 @@ private static boolean getProductInfoFromJar(String pathEntry, Map<String, Attri
...
+        int digits;
+
+        public GatewayProductVersionType(int digits) {
+            this.digits = digits;
+        }
+
+        private String getGatewayVersionType(String v) {
```
Comment (DoruM) [inline server.api/src/main/java/org/kaazing/gateway/server/impl/VersionUtils.java]: This should be a `toString` method.

File: server.api/src/main/java/org/kaazing/gateway/server/impl/VersionUtils.java [hunk 3]
```diff
@@ -238,4 +203,35 @@ private static boolean getProductInfoFromJar(String pathEntry, Map<String, Attri
         }
         return result;
     }
+
+    private static class GatewayProductVersionType {
+        int digits;
```
Comment (DoruM) [inline server.api/src/main/java/org/kaazing/gateway/server/impl/VersionUtils.java]: I believe it is better if you use title, version, edition, dependencies as membersi nstead of just digits.

File: server.api/src/main/java/org/kaazing/gateway/server/impl/VersionUtils.java [hunk 4]
```diff
@@ -31,10 +31,10 @@
 
 public final class VersionUtils {
 
-    private static String productTitle;
-    private static String productVersion;
-    private static String productEdition;
-    private static String productDependencies;
+    private static volatile String productTitle;
```
Comment (DoruM) [inline server.api/src/main/java/org/kaazing/gateway/server/impl/VersionUtils.java]: These should be members of the class ProductInfo (see below).

File: update.check/src/test/java/org/kaazing/gateway/update/check/UpdateCheckWithRCVersionIT.java [hunk 5]
```diff
@@ -41,7 +43,14 @@
         {
             GatewayConfiguration configuration = createGatewayConfiguration();
             //The product information from the server JAR will be overridden by the call to reset
-            VersionUtils.reset("Gateway", "Kaazing WebSocket Gateway", "5.6.1-RC001", "");
+           // ProductInfoReader.reset("Gateway", "Kaazing WebSocket Gateway", "5.6.1-RC001", "");
+            ProductInfo p = new ProductInfo();
+            p.setEdition("Gateway");
+            p.setTitle("Kaazing WebSocket Gateway");
+            p.setVersion("5.6.1-RC001");
```
Comment (sbadugu) [inline update.check/src/test/java/org/kaazing/gateway/update/check/UpdateCheckWithRCVersionIT.java:50]: @DoruM (CC: @dpwspoon) 
I just wanted to understand why are we hardcoding `5.6.1-RC001` version here, or for that matter why would we hard code any version?

Comment (dpwspoon) [reply update.check/src/test/java/org/kaazing/gateway/update/check/UpdateCheckWithRCVersionIT.java:50]: It looks like it is just a test, specifically to make sure things don't break when it has an RC version

File: server.api/src/test/java/org/kaazing/gateway/server/impl/ProductInfoReaderTest.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        Assert.assertEquals("5.0.0.8 Beta", productInfo.getVersion());
+        Assert.assertEquals("Kaazing WebSocket Gateway", productInfo.getTitle());
+        Assert.assertEquals("Test.Gateway", productInfo.getEdition());
+        Assert.assertEquals("0", productInfo.getMinor());
+        Assert.assertEquals("5", productInfo.getMajor());
+        Assert.assertEquals("0", productInfo.getPatch());
+        Assert.assertEquals("8 Beta", productInfo.getBuild());
+    }
+
+    @Test public void shouldGetDefaultProductInfo() {
```
Comment (dpwspoon) [inline server.api/src/test/java/org/kaazing/gateway/server/impl/ProductInfoReaderTest.java:41]: Nit: Please put @Test and other annotations above methods

**Pull request merged**